cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -15,46 +15,368 @@
+      transform_reduce_v_frontier_outgoing_e_by_dst(handle,
+                                                    graph_view,
+                                                    vertex_frontier.bucket(bucket_idx_cur),
+                                                    src_sigma.view(),
+                                                    dst_distance.view(),
+                                                    cugraph::edge_dummy_property_t{}.view(),
+                                                    brandes_e_op_t<vertex_t>{},
+                                                    reduce_op::plus<edge_t>());
+
+    update_v_frontier(handle,
+                      graph_view,
+                      std::move(new_frontier),
+                      std::move(new_sigma),
+                      vertex_frontier,
+                      std::vector<size_t>{bucket_idx_next},
+                      thrust::make_zip_iterator(distance.begin(), sigma.begin()),
+                      thrust::make_zip_iterator(distance.begin(), sigma.begin()),
+                      [hop] __device__(auto v, auto old_values, auto v_sigma) {
+                        vertex_t next_distance = thrust::get<0>(old_values) == invalid_distance
+                                                   ? (hop + 1)
+                                                   : thrust::get<0>(old_values);
+                        return thrust::make_tuple(
+                          thrust::make_optional(bucket_idx_next),
+                          thrust::make_optional(thrust::make_tuple(next_distance, v_sigma)));
+                      });
+
+    vertex_frontier.bucket(bucket_idx_cur).clear();
+    vertex_frontier.bucket(bucket_idx_cur).shrink_to_fit();
+    vertex_frontier.swap_buckets(bucket_idx_cur, bucket_idx_next);
+    if (vertex_frontier.bucket(bucket_idx_cur).aggregate_size() == 0) { break; }
+
+    ++hop;
+  }
+
+  return std::make_tuple(std::move(distance), std::move(sigma));
+}
+
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+void accumulate_vertex_results(
+  raft::handle_t const& handle,
+  graph_view_t<vertex_t, edge_t, false, multi_gpu> const& graph_view,
+  std::optional<edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+  rmm::device_uvector<weight_t>& centralities,
+  rmm::device_uvector<vertex_t>&& distance,
+  rmm::device_uvector<vertex_t>&& sigma,
+  bool with_endpoints,
+  bool do_expensive_check)
+{
+  constexpr vertex_t invalid_distance = std::numeric_limits<vertex_t>::max();
+
+  vertex_t diameter = thrust::transform_reduce(
+    handle.get_thrust_policy(),
+    distance.begin(),
+    distance.end(),
+    [] __device__(auto d) { return (d == invalid_distance) ? vertex_t{0} : d; },
+    vertex_t{0},
+    thrust::maximum<vertex_t>());
+
+  if constexpr (multi_gpu) {
+    diameter = host_scalar_allreduce(
+      handle.get_comms(), diameter, raft::comms::op_t::MAX, handle.get_stream());
+  }
+
```

Comment:
You can use transform_reduce_v to merge the two (assuming that distance.size() == graph_view.local_vertex_partition_range_size()).